src/screens/VaultScreen.js

```diff
@@ -13,11 +14,305 @@ export default class VaultScreen extends React.Component {
+  }
+  constructor(props) {
+    super(props);
+    this.state = {
+      titleName: "Resources",
+      searchText: '',
+      filter: [],
+      names: [
+        {name: 'CLEAR', selected:false, id: 1},
+        {name: 'FINANCE', selected:true, id: 2},
+        {name: 'HOUSING', selected:true, id: 3},
+        {name: 'EMPLOYMENT', selected:true, id: 4},
+        {name: 'LEGAL', selected:true, id: 5},
+        {name: 'MENTAL HEALTH', selected:true, id: 6},
+      ],
+      fontLoaded: false,
+      resources: [
+        {title: 'Title of Resource',
+          partner_org: 'Name of Partner Org',
+          date:'8 Oct 2018',
+          description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. In et aliquam neque. Proin lectus neque, tincidunt eget elementum sit amet, rhoncus ut libero.',
+          link: 'something',
+          upvotes: 0,
+          id: 1,
+        },
+        {title: 'Title of Resource',
+          partner_org: 'Name of Partner Org',
+          date:'8 Oct 2018',
+          description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. In et aliquam neque. Proin lectus neque, tincidunt eget elementum sit amet, rhoncus ut libero.',
+          link: 'something',
+          upvotes: 0,
+          id: 2,
+        },
+        {title: 'Title of Resource',
+          partner_org: 'Name of Partner Org',
+          date:'8 Oct 2018',
+          description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. In et aliquam neque. Proin lectus neque, tincidunt eget elementum sit amet, rhoncus ut libero.',
+          link: 'something',
+          upvotes: 0,
+          id: 3,
+        }
+      ]
+    };
+  }
+  updateFilter(item_id, new_state) {
+    var names_arr = this.state.names.slice()
+    for(var i = 0; i < names_arr.length; i++) {
+      if(names_arr[i].id == item_id) {
+        names_arr[i].selected = new_state;
+      }
+    };
+    this.setState({names:names_arr});
+  }
+  setOpposite(item_id) {
+    for(var i = 0; i < this.state.names.length; i++) {
+      if(this.state.names[i].id == item_id) {
+        this.updateFilter(item_id, !this.state.names[i].selected);
+      }
+    };
+  }
+  falseState(name, item_id) {
+    if(name=='Clear') {
+      for(var i = 1; i < this.state.names.length; i++) {
+        this.updateFilter(this.state.names[i].id, false)
+      }
+    } else {
+      this.setOpposite(item_id);
+    }
+  }
+  filterScroller() {
+    return this.state.names.map((item) => {
+      if (item.selected==false) {
+        return (
+          <TouchableHighlight key = {item.id} style={styles.item} onPress={() => {this.falseState(item.name, item.id);}}>
+            <Text style={{color:'white', fontSize:12, fontFamily: 'source-sans-pro-semibold',}}>{item.name}</Text>
+          </TouchableHighlight>
+        );
+      } else {
+        return (
+          <TouchableHighlight key = {item.id} style = {[styles.item, {backgroundColor:'white',}]} onPress={() => {this.setOpposite(item.id);}}>
```

Comment:
Again maybe this is a good place to write a class method and bind it before passing it into `onPress`